sdk/storage/azdatalake/datalakeerror/error_codes.go

```diff
@@ -0,0 +1,181 @@
+	MissingRequiredHeader                             Code = "MissingRequiredHeader"
+	MissingRequiredQueryParameter                     Code = "MissingRequiredQueryParameter"
+	MissingRequiredXMLNode                            Code = "MissingRequiredXmlNode"
+	MultipleConditionHeadersNotSupported              Code = "MultipleConditionHeadersNotSupported"
+	NoAuthenticationInformation                       Code = "NoAuthenticationInformation"
+	NoPendingCopyOperation                            Code = "NoPendingCopyOperation"
+	OperationNotAllowedOnIncrementalCopyBlob          Code = "OperationNotAllowedOnIncrementalCopyBlob"
+	OperationTimedOut                                 Code = "OperationTimedOut"
+	OutOfRangeInput                                   Code = "OutOfRangeInput"
+	OutOfRangeQueryParameterValue                     Code = "OutOfRangeQueryParameterValue"
+	PendingCopyOperation                              Code = "PendingCopyOperation"
+	PreviousSnapshotCannotBeNewer                     Code = "PreviousSnapshotCannotBeNewer"
+	PreviousSnapshotNotFound                          Code = "PreviousSnapshotNotFound"
+	PreviousSnapshotOperationNotSupported             Code = "PreviousSnapshotOperationNotSupported"
+	RequestBodyTooLarge                               Code = "RequestBodyTooLarge"
+	RequestURLFailedToParse                           Code = "RequestUrlFailedToParse"
+	ResourceAlreadyExists                             Code = "ResourceAlreadyExists"
+	ResourceNotFound                                  Code = "ResourceNotFound"
+	ResourceTypeMismatch                              Code = "ResourceTypeMismatch"
+	SequenceNumberConditionNotMet                     Code = "SequenceNumberConditionNotMet"
+	SequenceNumberIncrementTooLarge                   Code = "SequenceNumberIncrementTooLarge"
+	ServerBusy                                        Code = "ServerBusy"
+	SnapshotCountExceeded                             Code = "SnapshotCountExceeded"
+	SnapshotOperationRateExceeded                     Code = "SnapshotOperationRateExceeded"
+	SnapshotsPresent                                  Code = "SnapshotsPresent"
+	SourceConditionNotMet                             Code = "SourceConditionNotMet"
+	SystemInUse                                       Code = "SystemInUse"
+	TargetConditionNotMet                             Code = "TargetConditionNotMet"
+	UnauthorizedBlobOverwrite                         Code = "UnauthorizedBlobOverwrite"
+	UnsupportedHTTPVerb                               Code = "UnsupportedHttpVerb"
+	UnsupportedHeader                                 Code = "UnsupportedHeader"
+	UnsupportedQueryParameter                         Code = "UnsupportedQueryParameter"
+	UnsupportedXMLNode                                Code = "UnsupportedXmlNode"
+)
+
+var (
+	// MissingSharedKeyCredential - Error is returned when SAS URL is being created without SharedKeyCredential.
+	MissingSharedKeyCredential = bloberror.MissingSharedKeyCredential
+)
```

Comment:
The error codes overlap across the packages, would it be beneficial to make the error_codes exist in one spot that all the packages can use?

Reply:
we have discussed the possibility of a common package amongst our services, but we have not committed to that yet.